README.md

```diff
@@ -34,13 +34,20 @@ pip install siobrultech-protocols
 
 ```python
 import functools
-from siobrultech_protocols.gem.protocols import PacketProtocol
+from siobrultech_protocols.gem.protocols import PacketProtocol, PacketProtocolMessageType
 
 # Queue to get received packets from.
 queue = asyncio.Queue()
 
 # Pass this Protocol to whatever receives data from the device.
 protocol_factory = functools.partial(PacketProtocol, queue=queue)
+
+# Dequeue and look for packet received messages. (Typically do this in a loop.)
+message = await queue.get()
+if message.type == PacketProtocolMessageType.PacketReceived:
+    assert message.packet
```

Comment:
idle thought: if we made subclasses for this message type that had a non-optional packet, and we assert it's that type, would this be more clear?

Reply:
Another approach would be to skip the enum entirely, instead create a subclass per message type, and do `isinstance` checks. Thoughts? (I like that better from a typing standpoint, but I am not sure how Pythonic it is.)

Reply:
I don't have a strong opinion, to be honest.  Maybe it's at least worth adding a message to this assert in the readme so folks know why we are asserting it.

Reply:
Updated to use subclasses. I think it's cleaner overall. Good with you?